cpp/dolfin/fem/dofs_permutation.cpp

```diff
@@ -83,6 +91,89 @@ std::array<int, 4> calculate_tetrahedron_orders(int v1, int v2, int v3, int v4)
       "Two of a tetrahedron's vertices appear to be equal.");
 }
 //-----------------------------------------------------------------------------
+/// Calculates the number of times the rotation and reflection of a
+/// quadrilateral should be applied to a quadrilateral with the given global
+/// vertex numbers
+/// @param[in] v1 The global vertex number of the quadrilateral's first vertex
+/// @param[in] v2 The global vertex number of the quadrilateral's second vertex
+/// @param[in] v3 The global vertex number of the quadrilateral's third vertex
+/// @param[in] v4 The global vertex number of the quadrilateral's fourth vertex
+/// @return The rotation and reflection orders for the quadrilateral
+std::array<int, 2> calculate_quadrilateral_orders(int v1, int v2, int v3,
+                                                  int v4)
+{
+  if (v1 < v2 and v1 < v3 and v1 < v4)
+    return {0, v2 > v3};
+  else if (v2 < v1 and v2 < v3 and v2 < v4)
+    return {1, v4 > v1};
+  else if (v4 < v1 and v4 < v2 and v4 < v3)
+    return {2, v3 > v2};
+  else if (v3 < v1 and v3 < v2 and v3 < v4)
+    return {3, v1 > v4};
+
+  throw std::runtime_error(
+      "Two of a quadrilateral's vertices appear to be equal.");
+}
+//-----------------------------------------------------------------------------
+/// Calculates the number of times the rotations and reflection of a triangle
+/// should be applied to a hexahedron with the given global vertex numbers
+/// @param[in] v1 The global vertex number of the hexahedron's first vertex
+/// @param[in] v2 The global vertex number of the hexahedron's second vertex
+/// @param[in] v3 The global vertex number of the hexahedron's third vertex
+/// @param[in] v4 The global vertex number of the hexahedron's fourth vertex
+/// @param[in] v5 The global vertex number of the hexahedron's fifth vertex
+/// @param[in] v6 The global vertex number of the hexahedron's sixth vertex
+/// @param[in] v7 The global vertex number of the hexahedron's seventh vertex
+/// @param[in] v8 The global vertex number of the hexahedron's eighth vertex
```

Comment:
Wow this is verbose...

Reply:
Yes, I'll change it to "v1 to v8 are..."